rust/crd/src/storage.rs

```diff
@@ -0,0 +1,47 @@
+use stackable_operator::builder::{ContainerBuilder, PodBuilder, VolumeBuilder};
+use stackable_operator::config::merge::Merge;
+use stackable_operator::k8s_openapi::apimachinery::pkg::api::resource::Quantity;
+use stackable_operator::schemars::{self, JsonSchema};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{SC_DIRECTORY, SC_VOLUME_NAME};
+
+/// Storage configuration used by all roles except historical
+#[derive(Clone, Debug, Default, Deserialize, Eq, JsonSchema, Merge, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DruidStorage {}
+
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, Merge, PartialEq, Serialize, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct HistoricalStorage {
+    pub segment_cache_size_gb: Option<u16>,
```

Comment:
Yeah but then you open a Pandora's box of unit conversions, handling fractions, etc. All of this needs to be implemented in central place since `Quantity` is just a `String` wrapper and it seems like overkill to do it here.

Reply:
I'm okay with that :+1:  resolving this.